sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/options/BlobParallelUploadOptions.java

```diff
@@ -103,11 +126,23 @@ public InputStream getDataStream() {
      *
      * @return The exact length of the data. It is important that this value match precisely the length of the
      * data provided in the {@link InputStream}.
+     * @deprecated use {@link #getOptionalLength()} to have safe access to a length that will not always exist.
      */
+    @Deprecated
     public long getLength() {
         return length;
```

Comment:
is it worth saving the customer of a NPE here?

Reply:
Or if length is null should we throw a meaningful error?

Reply:
I think this is fine, it won't throw if they haven't switched to new code path.